packages/squiggle-lang/src/rescript/Reducer/Reducer_Category/Reducer_Category_Module_TypeChecker.res

```diff
@@ -0,0 +1,21 @@
+module ExpressionValue = ReducerInterface_InternalExpressionValue
+module ExpressionT = Reducer_Expression_T
+open ExpressionValue
+
+let isOfResolvedType = (aType, aValue) => {
+  let caseTypeIdentifier = (aTypeIdentifier0, aValue) => {
+    let valueType = aValue->valueToValueType->valueTypeToString->Js.String2.toLowerCase
+    let aTypeIdentifier = aTypeIdentifier0->Js.String2.toLowerCase
+    aTypeIdentifier === valueType
+  }
+
+  switch aType {
+  | IEvTypeIdentifier(aTypeIdentifier) => caseTypeIdentifier(aTypeIdentifier, aValue)
+  | _ => false
+  }
+}
+
+// let compileTypeExpression = (typeExpression: string, bindings: ExpressionT.bindings, reducerFn: ExpressionT.reducerFn) => {
+//     statement = `type compiled=${typeExpression}`
```

Comment:
What's with these bits? Is this meant to stay here, for a future story? Generally I don't want to see much commented-out code, it's not clear if it's dead or not.